Instead of checking specifically for ports' installed libiconv on FreeBSD
systems, with this patch the AM_ICONV macro from GNU gettext is used, that
checks in a generic way for iconv's presence. Allows to work fine on non
FreeBSD, non x86 and non ports systems.

Index: xine-lib-1.1.1/configure.ac
===================================================================
--- xine-lib-1.1.1.orig/configure.ac
+++ xine-lib-1.1.1/configure.ac
@@ -169,6 +169,7 @@ if test x"$USE_INCLUDED_LIBINTL" = x"yes
 fi
 AC_SUBST(INTLDIR)
 
+AM_ICONV
 
 dnl ---------------------------------------------
 dnl Checks for typedefs, structures, and compiler characteristics.
@@ -593,22 +594,6 @@ case "$host_or_hostalias" in
 	fi
 	;;
 
-  i?86-*-freebsd*)
-	AC_CHECK_HEADER(/usr/local/include/iconv.h,
-			CFLAGS="-I/usr/local/include $CFLAGS"
-			LDFLAGS="-L/usr/local/lib $LDFLAGS"
-	                LIBICONV="-liconv"
-                        [ echo
-                          echo "****************************************************************"
-		          echo "* You need to install a recent version of the port 'libiconv'  *"
-			  echo "* (in /usr/ports/converters/libiconv).                         *"
-			  echo "* The library in this port is needed to successfully compile   *"
-			  echo "* libsputext plugin.                                           *"
-                          echo "****************************************************************"
-	                  echo 
-                        ],)
-        ;;
-
   ppc-*-linux* | powerpc-*)
         ppc_arch="yes"
         ;;
